specification/purview/data-plane/Azure.Analytics.Purview.Insights/preview/2022-03-01-preview/purviewinsights.json

```diff
@@ -0,0 +1,1202 @@
+    },
+    "DeletedAssetsDistributionSnapshotRecord": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/SnapshotRecord"
+        }
+      ],
+      "properties": {
+        "classification": {
+          "type": "string"
+        },
+        "classificationCategory": {
+          "type": "string"
+        },
+        "collectionName": {
+          "type": "string"
+        },
+        "count": {
+          "format": "int64",
+          "type": "integer"
+        },
+        "dataOwnerAssigned": {
+          "type": "boolean"
+        }
+      },
+      "x-ms-discriminator-value": "DeletedAssetsDistribution"
+    },
+    "FilesWithoutResourceSetSnapshotRecord": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/SnapshotRecord"
+        }
+      ],
+      "properties": {
+        "sourceType": {
+          "type": "string"
+        },
+        "collectionName": {
+          "type": "string"
+        },
+        "path": {
+          "type": "string"
+        },
+        "count": {
+          "format": "int64",
+          "type": "integer"
+        },
+        "resourceSetCount": {
+          "format": "int64",
+          "type": "integer"
+        }
+      },
+      "x-ms-discriminator-value": "FilesWithoutResourceSet"
+    },
+    "NetNewAssetsDistributionSnapshotRecord": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/SnapshotRecord"
+        }
+      ],
+      "properties": {
+        "classification": {
+          "type": "string"
+        },
+        "classificationCategory": {
+          "type": "string"
+        },
+        "collectionName": {
+          "type": "string"
+        },
+        "count": {
+          "format": "int64",
+          "type": "integer"
+        },
+        "dataOwnerAssigned": {
+          "type": "boolean"
+        }
+      },
+      "x-ms-discriminator-value": "NetNewAssetsDistribution"
+    }
+  },
+  "parameters": {
+    "x-ms-account-name": {
+      "in": "header",
+      "name": "x-ms-account-name",
+      "description": "The Account Id",
+      "type": "string",
+      "x-ms-client-name": "AccountName"
+    },
+    "x-ms-account-id": {
+      "in": "header",
+      "name": "x-ms-account-id",
+      "description": "The Account Id",
+      "type": "string",
+      "x-ms-client-name": "AccountId"
+    },
+    "api-version": {
+      "in": "query",
+      "name": "api-version",
+      "description": "The api version to use.",
+      "required": true,
+      "type": "string",
+      "x-ms-client-name": "ApiVersion"
+    }
+  },
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "flow": "implicit",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      },
+      "description": "Azure Active Directory OAuth2 Flow."
+    }
+  }
```

Comment:
You should also specify a global [`security` requirement](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md#security-requirement-object) that specifies the "azure_auth" security scheme.

Reply:
[Purview Scanning API](https://github.com/Azure/azure-rest-api-specs/blob/5dad3b83d0926768ede15ed41895d5f38defd512/specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2021-10-01-preview/scanningService.json#L6926) appears the same in how azure_auth is specified on 1192. How else is this done?

Reply:
The "azure_auth" definition is fine, but it just defines the security scheme. The "security requirement" defines which operations use that scheme.  A securityDefinition is like a schema in the "definitions" section -- it does nothing until it is actually referenced, and the security requirement is how it is referenced.  The security requirement can be global, which means it applies to all operations (unless overridden).  It can also be defined at the operation level, but I think a global one is probably what you want.

Here's a good example of what that should look like: [deviceupdate.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/deviceupdate/data-plane/Microsoft.DeviceUpdate/preview/2021-06-01-preview/deviceupdate.json#L26)

Reply:
Thank you, makes sense. Updated